src/phoenix/experimental/evals/functions/binary.py

```diff
@@ -100,24 +115,30 @@ def run_relevance_eval(
     ].map(list)
 
     exploded_df = filtered_df.explode(retrieved_documents_column_name, ignore_index=False)
-    exploded_df[retrieved_document_text_column_name] = [
-        document_data["document.content"] if document_data is not None else None
-        for document_data in exploded_df[retrieved_documents_column_name]
-    ]
+    if trace_data:
+        exploded_df[retrieved_document_text_column_name] = [
+            document_data["document.content"] if document_data is not None else None
+            for document_data in exploded_df[retrieved_documents_column_name]
```

Comment:
this is not pure - it's mutating the dataframe - this is gonna cause problems because the user might not be expecting that data to be modified and when the data is grouped back the data will lose score and document IDs.

Reply:
This approach wasn't really put in by me, just matched what was being done or is in the code now. 

If you look further down, its' grouped back up and inserted into a column in the original dataframe, so I don't think you lose anything. 

Might be cleaner ways of doing this. 

    collapsed_df = exploded_df.groupby(exploded_df.index, axis="index").agg(
        {
            llm_relevance_column_name: list,
        }
    )
    output_df = pd.DataFrame(index=dataframe.index)
    output_df[llm_relevance_column_name] = None
    output_df.loc[collapsed_df.index, llm_relevance_column_name] = collapsed_df[
        llm_relevance_column_name
    ]
    return output_df[llm_relevance_column_name].tolist()